runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/EngineContextTest.java

```diff
@@ -0,0 +1,103 @@
+ * under the License.
+ */
+package io.aklivity.zilla.runtime.engine;
+
+import static io.aklivity.zilla.runtime.engine.EngineConfiguration.ENGINE_WORKERS;
+import static io.aklivity.zilla.runtime.engine.internal.layouts.Layout.Mode.CREATE_READ_WRITE;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.rules.RuleChain.outerRule;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.function.LongConsumer;
+import java.util.function.LongSupplier;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.DisableOnDebug;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+
+import io.aklivity.zilla.runtime.engine.internal.LabelManager;
+import io.aklivity.zilla.runtime.engine.internal.layouts.metrics.CountersLayout;
+import io.aklivity.zilla.runtime.engine.internal.stream.NamespacedId;
+import io.aklivity.zilla.runtime.engine.test.EngineRule;
+
+public class EngineContextTest
+{
+    private static final String ENGINE_DIRECTORY = "target/zilla-itests-ati1";
+
+    private final TestRule timeout = new DisableOnDebug(new Timeout(10, SECONDS));
+
+    private final EngineRule engine = new EngineRule()
+        .directory(ENGINE_DIRECTORY)
+        .configure(ENGINE_WORKERS, 3)
+        .commandBufferCapacity(8192)
+        .responseBufferCapacity(8192)
+        .counterValuesBufferCapacity(8192)
+        .clean();
+
+    @Rule
+    public final TestRule chain = outerRule(engine).around(timeout);
+
+    @Test
+    public void shouldFetchCounterValue() throws Exception
+    {
+        // GIVEN
+        // create labels
+        LabelManager labels = new LabelManager(Paths.get(ENGINE_DIRECTORY));
+        int namespaceId = labels.supplyLabelId("ns1");
+        int bindingId = labels.supplyLabelId("binding1");
+        int metricId = labels.supplyLabelId("counter1");
+        long namespacedBindingId = NamespacedId.id(namespaceId, bindingId);
+        long namespacedMetricId = NamespacedId.id(namespaceId, metricId);
+
+        // write some metric data
+        Path path0 = Paths.get(ENGINE_DIRECTORY, "metrics", "counters0");
+        Path path1 = Paths.get(ENGINE_DIRECTORY, "metrics", "counters1");
+        Path path2 = Paths.get(ENGINE_DIRECTORY, "metrics", "counters2");
+        CountersLayout countersLayout0 = new CountersLayout.Builder()
+            .path(path0)
+            .capacity(8192)
+            .mode(CREATE_READ_WRITE)
+            .build();
+        CountersLayout countersLayout1 = new CountersLayout.Builder()
+            .path(path1)
+            .capacity(8192)
+            .mode(CREATE_READ_WRITE)
+            .build();
+        CountersLayout countersLayout2 = new CountersLayout.Builder()
+            .path(path2)
+            .capacity(8192)
+            .mode(CREATE_READ_WRITE)
+            .build();
```

Comment:
The `labels` and `counters` layouts should already be created by the `engine`, right?

Perhaps we just need an `@Configuration` `zilla.yaml` that registers a `test` metric against a `test` binding and a method on `EngineRule` to supply the counter writer for a specific core# ?